lib/BlockedBy.js

```diff
@@ -133,9 +134,11 @@ class BlockedBy extends NodeResque.Plugin {
         const enableCollapse = String(this.options.collapse) === 'true'; // because kubernetes value is a string
         const buildConfig = await this.queueObject.connection.redis.hget(
             `${queuePrefix}buildConfigs`, buildId);
-        const collapse = hoek.reach(JSON.parse(buildConfig),
-            'annotations>screwdriver.cd/collapseBuilds',
-            { default: enableCollapse, separator: '>' });
+        const annotations = hoek.reach(JSON.parse(buildConfig), 'annotations', { default: {} });
+        const collapse = hoek.reach(annotations, 'screwdriver.cd/collapseBuilds', {
+            default: enableCollapse, separator: '>' });
+        const timeout = hoek.reach(annotations, 'screwdriver.cd/timeout', { separator: '>' });
+        const blockTimeout = timeout ? timeout + BLOCK_TIMEOUT_BUFFER : this.blockTimeout();
```

Comment:
pass the `timeout` to `blockTimeout` function and handle it there ? It's anyways abstracting some logic, so why not handle everything there.

Reply:
makes sense, will fix it